nexa/gguf/streamlit/streamlit_vlm.py

```diff
@@ -1,40 +1,108 @@
 import sys
 import tempfile
-from typing import Iterator
-
+import subprocess
+import re
+from typing import List, Iterator
 import streamlit as st
 from PIL import Image
 from nexa.general import pull_model
 from nexa.gguf.nexa_inference_vlm import NexaVLMInference
+from nexa.constants import NEXA_RUN_MODEL_MAP_VLM
 
-default_model = sys.argv[1]
-is_local_path = False if sys.argv[2] == "False" else True
-hf = False if sys.argv[3] == "False" else True
-projector_local_path = sys.argv[4] if len(sys.argv) > 4 else None
+# init from command line args:
+try:
+    default_model = sys.argv[1]
+    is_local_path = sys.argv[2].lower() == "true"
+    hf = sys.argv[3].lower() == "true"
+    projector_local_path = sys.argv[4] if len(sys.argv) > 4 else None
+except IndexError:
+    st.error("Missing required command line arguments.")
+    sys.exit(1)  # program terminated with an error
 
+def parse_nexa_list_output() -> List[str]:
```

Comment:
same issue, do not parse string
also for general functions, put them in utils.py and import, do not copy code blocks multiple times.